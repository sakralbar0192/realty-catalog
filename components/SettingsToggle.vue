<template>
  <button
    :class="styles.toggleButton"
    @click="togglePanel"
    :aria-label="$t('settings.open')"
    :aria-expanded="isOpen"
    data-test="settings-toggle"
  >
    <Icon name="settings" :size="20" />
  </button>
</template>

<script setup lang="ts">
import { inject, computed, type ComputedRef } from 'vue'
import Icon from '~/components/Icon.vue'
import { useAppI18n } from '~/composables/useI18n'
import styles from '~/assets/styles/components/SettingsToggle.module.scss'

const { translate: $t } = useAppI18n()

// Inject the settings panel methods
const settingsPanel = inject('settingsPanel') as ComputedRef<{
  open: () => void
  close: () => void
  isOpen: boolean
}> | null

const isOpen = computed(() => settingsPanel?.value?.isOpen ?? false)

const togglePanel = () => {
  if (settingsPanel?.value) {
    if (settingsPanel.value.isOpen) {
      settingsPanel.value.close()
    } else {
      settingsPanel.value.open()
    }
  }
}
</script>