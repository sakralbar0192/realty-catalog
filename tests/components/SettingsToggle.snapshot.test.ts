import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { computed } from 'vue'
import SettingsToggle from '../../components/SettingsToggle.vue'

// Mock the useAppI18n composable
vi.mock('../../composables/useI18n', () => ({
  useAppI18n: () => ({
    translate: (key: string) => {
      const translations: Record<string, string> = {
        'settings.open': 'Open settings',
      }
      return translations[key] || key
    },
  }),
}))

describe('SettingsToggle Snapshots', () => {
  it('should match snapshot when panel is closed', () => {
    const mockSettingsPanel = {
      open: vi.fn(),
      close: vi.fn(),
      isOpen: false,
    }

    const wrapper = mount(SettingsToggle, {
      global: {
        provide: {
          settingsPanel: computed(() => mockSettingsPanel),
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot when panel is open', () => {
    const mockSettingsPanel = {
      open: vi.fn(),
      close: vi.fn(),
      isOpen: true,
    }

    const wrapper = mount(SettingsToggle, {
      global: {
        provide: {
          settingsPanel: computed(() => mockSettingsPanel),
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
