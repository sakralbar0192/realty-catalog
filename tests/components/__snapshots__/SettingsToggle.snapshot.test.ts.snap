// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsToggle Snapshots > should match snapshot when panel is closed 1`] = `"<button class="_toggleButton_daa333" aria-label="Open settings" aria-expanded="false" data-test="settings-toggle"><span class="_icon_0599e9 icon--20"><svg width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="12" cy="12" r="3"></circle><path d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1 1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"></path></svg></span></button>"`;

exports[`SettingsToggle Snapshots > should match snapshot when panel is open 1`] = `"<button class="_toggleButton_daa333" aria-label="Open settings" aria-expanded="true" data-test="settings-toggle"><span class="_icon_0599e9 icon--20"><svg width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="12" cy="12" r="3"></circle><path d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1 1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"></path></svg></span></button>"`;
